src/Microsoft.AspNetCore.Routing/Matchers/MatcherEndpoint.cs

```diff
@@ -44,6 +46,9 @@ public sealed class MatcherEndpoint : Endpoint
             RequiredValues = requiredValues;
             var mergedDefaults = GetDefaults(ParsedTemplate, defaults);
             Defaults = mergedDefaults;
+
+            var mergedReferences = MergeMatchProcessorReferences(ParsedTemplate, nonInlineMatchProcessorReferences);
```

Comment:
Will this change with https://github.com/aspnet/Routing/pull/585?

Reply:
I kept this here so that we can be consistent with how `Defaults` are handled here. If we change for #585, both `Defaults` and `MatchProcessorReferences` would need to change.